metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -940,6 +943,294 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
+            #         id
+            #         __typename
+            #       }
+            #       embeddedDatasources {
+            #         id
+            #         __typename
+            #       }
+            #       __typename
+            #     }
+            #     pageInfo {
+            #       hasNextPage
+            #       endCursor
+            #       __typename
+            #     }
+            #     __typename
+            #   }
+            # } Variables: {"first": 100, "after": null}
+            # [2025-01-30 10:12:28,669] INFO     {datahub.ingestion.source.tableau.tableau:969} - [[TURMS]] Result: {'data': {'workbooksConnection': {'nodes': [{'id': '280d3fd0-c431-30c3-2e8f-c2620a4f472e', 'name': 'customer_profit_wb', 'luid': '3c6b22b8-5c42-413f-9505-f5d4715413db', 'uri': 'sites/71413/workbooks/2782740', 'projectName': 'project_test2', 'owner': {'username': 'harshal@acryl.io', '__typename': 'TableauUser'}, 'description': '', 'createdAt': '2024-09-03T04:36:19Z', 'updatedAt': '2024-09-03T04:36:19Z', 'tags': [], 'sheets': [{'id': 'aa141b11-a1ae-fef6-4a25-28b85d62b3f4', '__typename': 'Sheet'}], 'dashboards': [], 'embeddedDatasources': [{'id': 'a00e6ea7-64f5-f999-9967-236609717f86', '__typename': 'EmbeddedDatasource'}], '__typename': 'Workbook'}], 'pageInfo': {'hasNextPage': False, 'endCursor': None, '__typename': 'PageInfo'}, '__typename': 'WorkbooksConnection'}}}
+            # [2025-01-30 10:12:28,669] INFO     {datahub.ingestion.source.tableau.tableau:971} - [[TURMS]] Workbooks: workbooks_connection=GetItems_workbooksConnectionWorkbooksconnection(typename='WorkbooksConnection', nodes=[GetItems_workbooksConnectionWorkbooksconnectionNodes(typename='Workbook', id='280d3fd0-c431-30c3-2e8f-c2620a4f472e', name='customer_profit_wb', luid='3c6b22b8-5c42-413f-9505-f5d4715413db', uri='sites/71413/workbooks/2782740', project_name='project_test2', owner=GetItems_workbooksConnectionWorkbooksconnectionNodesOwner(typename='TableauUser', username='harshal@acryl.io'), description='', created_at=datetime.datetime(2024, 9, 3, 4, 36, 19, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2024, 9, 3, 4, 36, 19, tzinfo=datetime.timezone.utc), tags=[], sheets=[GetItems_workbooksConnectionWorkbooksconnectionNodesSheets(typename='Sheet', id='aa141b11-a1ae-fef6-4a25-28b85d62b3f4')], dashboards=[], embedded_datasources=[GetItems_workbooksConnectionWorkbooksconnectionNodesEmbeddeddatasources(typename='EmbeddedDatasource', id='a00e6ea7-64f5-f999-9967-236609717f86')])], page_info=GetItems_workbooksConnectionWorkbooksconnectionPageinfo(typename='PageInfo', has_next_page=False, end_cursor=None))
+
+            # SGQLC
+
+            # mypy errors
+            #
+            # src/datahub/ingestion/source/tableau/codegen_sgqlc/tableau_schema.py:37553: error: Incompatible types in assignment (expression has type tuple[str, str, str, str], base class "DataField" defined the type as tuple[str, str, ... <34 more items>])  [assignment]
```

Comment:
The mypy error here is because `BinField` redefines `__field_names__` from parent class. And same patter for all mypy errors below.

```
class DataField(sgqlc.types.Interface):
    """Base GraphQL type for a field containing data. Most types of
    Fields will implement this interface with exceptions like
    HierarchyField.
    """
    
    __schema__ = tableau_schema
    __field_names__ = (
        "data_category",
        "data_type",
        ...,
    )
    ...

class BinField(sgqlc.types.Type, DataField, Field, FieldReferencingField, Node):
    __schema__ = tableau_schema
    __field_names__ = ("bin_size", "formula", "parameters", "parameters_connection")
    ...
```

Beyond this mypy errors, no manual updates were required here to make this work.